src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,158 @@
+  }
+
+  let accordion;
+  if (props.onChange === undefined) {
+    accordion = (
+      <div
+        className={styleClasses + ' ' + isExpanded}
+        data-testid="accordion-element"
+      >
+        <div className={styleClasses + ' accordion-title-container'}>
+          {type === 'checkbox' && (
+            <input
+              className={styleClasses + ' accordion-title-checkbox'}
+              type={type}
+            />
+          )}
+          <button
+            className={styleClasses + ' accordion-title'}
+            onClick={handleDefaultClick}
+            disabled={disabled}
+          >
+            <div className={styleClasses + ' accordion-title-summary'}>
+              <p className={styleClasses + ' accordion-title-information'}>
+                {title}
+              </p>
+              <p className={styleClasses + ' accordion-title-additional'}>
+                {summary}
+              </p>
+              <img
+                className={styleClasses + ' accordion-title-icon'}
+                src={isExpanded ? props.retractIcon : props.expandIcon}
+                alt="mySvgImage"
+              />
+            </div>
+          </button>
+        </div>
+
+        {show ? (
+          <p className={styleClasses + ' accordion-description'}>
+            {props.children}
+          </p>
+        ) : (
+          <p className={styleClasses + ' accordion-description hidden'}>
+            {props.children}
+          </p>
+        )}
+      </div>
+    );
+  } else {
+    accordion = (
+      <div
+        className={styleClasses + ' ' + isControlExpanded}
+        data-testid="accordion-element"
+      >
+        <div className={styleClasses + ' accordion-title-container'}>
+          {type === 'checkbox' && (
+            <input
+              className={styleClasses + ' accordion-title-checkbox'}
+              type={type}
+            />
+          )}
+          <button
+            className={styleClasses + ' accordion-title'}
+            onClick={onChange}
+            disabled={disabled}
+          >
+            <div className={styleClasses + ' accordion-title-summary'}>
+              <p className={styleClasses + ' accordion-title-information'}>
+                {title}
+              </p>
+              <p className={styleClasses + ' accordion-title-additional'}>
+                {summary}
+              </p>
+              <img
+                className={styleClasses + ' accordion-title-icon'}
+                src={isExpanded ? props.retractIcon : props.expandIcon}
+                alt="mySvgImage"
+              />
+            </div>
+          </button>
+        </div>
+
+        {props.expanded ? (
+          <p className={styleClasses + ' accordion-description'}>
+            {props.children}
+          </p>
+        ) : (
+          <p className={styleClasses + ' accordion-description hidden'}>
+            {props.children}
+          </p>
+        )}
+      </div>
+    );
+  }
+  return accordion;
```

Comment:
So now you can simply just return the JSX, instead of extract the logic into a variable with if statement. Also I realized that you used a lot of props.xxx, but you actually de-structure most of them at the beginning of your component. It seems redundant to me.